File: sorting_&_searching/go_/bubble_sort.go
package main

import "fmt"

func BubbleSort(arr []int) []int {
	for i := 0; i < len(arr); i++ {
		for j := i + 1; j < len(arr); j++ {
			if arr[i] > arr[j] {
				temp := arr[i]
				arr[i] = arr[j]
				arr[j] = temp
			}
		}
	}
	return arr
}

func main() {
	arr := []int{5, 3, 2, 1}
	fmt.Println(BubbleSort(arr))
}

File: sorting/bubble_sort.py

def bubble_sort(arr):
    length = len(arr)
    for i in range(length):
        for j in range(i+1,length):
            if arr[i] > arr[j]:
                arr[i],arr[j] = arr[j],arr[i]
    return arr

print(bubble_sort([6,5,4,2,2,1]))